torch/csrc/distributed/autograd/context/container.cpp

```diff
@@ -161,10 +161,13 @@ void DistAutogradContainer::sendReleaseContextRpc(int64_t context_id) {
   auto workerIds =
       autograd_context_.find(context_id)->second->getKnownWorkerIds();
   auto agent = rpc::RpcAgent::getCurrentRpcAgent();
+  rpc::RpcRetryOptions options;
+  options.maxRetries = 20;
   for (const auto& worker_id : workerIds) {
-    agent->send(
+    agent->sendWithRetries(
         agent->getWorkerInfo(worker_id),
-        CleanupAutogradContextReq(context_id).toMessage());
+        CleanupAutogradContextReq(context_id).toMessage(),
+        options);
```

Comment:
Shouldn't we attach a callback to the Future returned here and if there is a failure, LOG(ERROR) that we couldn't release the context?